cogeo_mosaic/backend/file.py

```diff
@@ -0,0 +1,46 @@
+import functools
+from typing import BinaryIO, Dict, Tuple
+
+import mercantile
+from boto3.session import Session as boto3_session
+
+from cogeo_mosaic.backend.base import BaseBackend
+from cogeo_mosaic.backend.utils import get_assets_from_json
+from cogeo_mosaic.utils import _decompress_gz
+
+
+class FileBackend(BaseBackend):
+    """Local File Backend Adapter"""
+
+    def __init__(self, path: str):
+        self.mosaic_def = self.fetch_mosaic_definition(path)
+
+    def metadata(self) -> Dict:
+        return self.mosaic_def
```

Comment:
I think it would be good to keep consistency between each backend. e.g dynamoDB will return only the metadata while here we will return the full mosaicJSON.

Reply:
This is true, but I'm not sure how to get around that. It's not possible to return the entire `tiles` key for DynamoDB, so either `tiles` is _sometimes_ returned, or _never_ returned?

Reply:
Also I think some of these default things should go in the base class, like the default
```py
    def metadata(self) -> Dict:
        return self.mosaic_def
```

Reply:
yeah tiles (or quadkeys) could be optional

Reply:
> Also I think some of these default things should go in the base class, like the default

yeah but again `metadata` should be for metadata not the full mosaicJSON IMO

Reply:
As it is it's optional; it exists in all classes other than DynamoDB. That doesn't conform to the MosaicJSON spec, so maybe create an empty `tiles` array?

Reply:
> That doesn't conform to the MosaicJSON spec, so maybe create an empty tiles array

Well it's not true that it doesn't conform to the spec, yes it's not a JSON docs but it must have the same info (metadata + quadkey base index)

> so maybe create an empty tiles array?
no just raise an exception 
```
raise NotAvailable("{} is not available for DynamoDB backend")
```

Reply:
I'm confused. You want to raise an error instead of returning _any_ metadata? 

I'd be in favor of returning all metadata except for `tiles`, and just have a note in the docs that a function using `Backend.metadata` can't depend on the `tiles` key existing.

Reply:
I'm resolving this here; lets continue discussion on the comment about `metadata` class method

Reply:
yeah that's what I mean, 
metadata should return the mosaic metadata (zoom, bounds ...) but not the quadkey list (tiles) 

this (tiles) should be covered by another method/properties which should raise an error in dynamoDB